### PR TITLE
 Automatically update docs with Travis 

### PR DESCRIPTION
**Pull request by gaborigloi**
Signed-off-by: Gabor Igloi <gabor.igloi@citrix.com>

### CODE REVIEW

File: update_gh_pages.sh [hunk 1]
```diff
@@ -1,10 +1,37 @@
-#!/bin/bash
+#!/bin/sh
```
Comment (mseri) [inline update_gh_pages.sh]: What about movin this script with the other travis tools?

File: update_gh_pages.sh [hunk 2]
```diff
@@ -1,10 +1,37 @@
...
+opam install -y caml2html
 make html
-rm -fr /tmp/doc
-rsync -av doc/gen/ /tmp/doc/
-git checkout gh-pages
-mv .git /tmp/doc/
-rsync --delete -av /tmp/doc/ .
 
+if [ -z "$TRAVIS" -o "$TRAVIS_PULL_REQUEST" != "false" -o "$TRAVIS_BRANCH" != "master" ]; then
+  echo "This is not a push Travis-ci build on the master branch, doing nothing..."
```
Comment (lindig) [inline update_gh_pages.sh]: Typo: s/build/built/. I would rephrase it: "This pull request does not target the master branch - ignoring it"

File: update_gh_pages.sh [hunk 3]
```diff
@@ -1,10 +1,37 @@
-#!/bin/bash
+#!/bin/sh
+
+# Make sure we're not echoing any sensitive data
+set +x
```
Comment (lindig) [inline update_gh_pages.sh]: Are you sure that this is not exposing environment variables? I believe `-x` exposes all variable content:

```
: tmp $ cat foo.sh 
#! /bin/sh

set -x

touch $X
: tmp $ env X=bar ./foo.sh 
+ touch bar
```

File: update_gh_pages.sh (hunk 1 above)
Comment (gaborigloi) [reply update_gh_pages.sh]: That sounds good. In fact, we only need to pass the doc output dir to it.

File: update_gh_pages.sh (hunk 3 above)
Comment (gaborigloi) [reply update_gh_pages.sh]: Yes, and I think the `+` in `set +x`  disables this `-x` flag.

File: update_gh_pages.sh (hunk 1 above)
Comment (gaborigloi) [reply update_gh_pages.sh]: How about merging this to see if it works, and if it does, I'll factor out the duplication into a script in xapi-travis-scripts ?

File: update_gh_pages.sh (hunk 2 above)
Comment (gaborigloi) [reply update_gh_pages.sh]: I think this is accurate, as this must not run for a pull request at all; only for builds running in Travis for commits pushed to the master branch.

Comment (gaborigloi) [reply update_gh_pages.sh]: I've removed the word `push`, now it sounds better.

File: update_gh_pages.sh (hunk 1 above)
Comment (mseri) [reply update_gh_pages.sh]: Sounds good to me
